### PR TITLE
[Instrument template] Remove each() as is deprecated since php 7.2 and removed in php 8.0

### LINKED ISSUE #8241: [Instrument template] each() is deprecated since php 7.2 and removed in php 8.0
https://github.com/aces/Loris/blob/1d3a4d0d971c0fa99323421fa89cf9c97b747294/docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc#L153

The each function is deprecated as of PHP 7.2.0, and REMOVED as of PHP 8.0.0. Relying on this function is highly discouraged.

### PR DESCRIPTION
**Pull request by shonibare**
fixes #8241

### CODE REVIEW

File: docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc [hunk 1]
```diff
@@ -150,7 +150,7 @@ class NDB_BVL_Instrument_TEST_NAME extends NDB_BVL_Instrument
         }
         //echo error messages
         if(!empty($file->errorLog)){
-        while(list($fileType,$fileErrors)=each($file->errorLog)){
+        foreach($file->errorLog AS $fileType => $fileErrors){
```
Comment (xlecours) [inline docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:153]: Please remove the `if` statement above. if it's empty, the foreach block will just run zero time.
Also fix indentation of curly brackets.

**Review by xlecours — CHANGES_REQUESTED** [review]

File: docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc [hunk 2]
```diff
@@ -149,13 +149,11 @@ class NDB_BVL_Instrument_TEST_NAME extends NDB_BVL_Instrument
...
         //echo error messages
-        if(!empty($file->errorLog)){
-        while(list($fileType,$fileErrors)=each($file->errorLog)){
-            foreach($fileErrors AS $error){
-                 echo "<span style='color:red'><b>Upload Error</b> $fileType: $error</span><br>";
-            }
-         }
-        }
+        foreach($file->errorLog AS $fileType => $fileErrors){
+                foreach($fileErrors AS $error){
```
Comment (xlecours) [inline docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:153]: indentation must be 4 spaces.

**Review by xlecours — CHANGES_REQUESTED** [review]

File: docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc (hunk 2 above)
Comment (laemtl) [reply docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:153]: Interesting! Why is this passing the php linter?

Comment (xlecours) [reply docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:153]: I don't think php linter look at those.

Comment (xlecours) [reply docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:153]: there is no docs in https://github.com/aces/Loris/blob/main/test/run-php-linter.sh#L63

File: docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc [hunk 3]
```diff
@@ -149,17 +149,15 @@ class NDB_BVL_Instrument_TEST_NAME extends NDB_BVL_Instrument
             $file->processFiles();
         }
         //echo error messages
-        if(!empty($file->errorLog)){
-        while(list($fileType,$fileErrors)=each($file->errorLog)){
+        foreach($file->errorLog AS $fileType => $fileErrors){
             foreach($fileErrors AS $error){
-                 echo "<span style='color:red'><b>Upload Error</b> $fileType: $error</span><br>";
+		    echo "<span style='color:red'><b>Upload Error</b> $fileType: $error</span><br>";
```
Comment (xlecours) [inline docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:154]: Indentation should be 4 spaces by level.

**Review by xlecours — CHANGES_REQUESTED** [review]

File: docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc [hunk 4]
```diff
@@ -149,17 +149,16 @@ class NDB_BVL_Instrument_TEST_NAME extends NDB_BVL_Instrument
             $file->processFiles();
         }
         //echo error messages
-        if(!empty($file->errorLog)){
-        while(list($fileType,$fileErrors)=each($file->errorLog)){
+        foreach($file->errorLog AS $fileType => $fileErrors){
             foreach($fileErrors AS $error){
-                 echo "<span style='color:red'><b>Upload Error</b> $fileType: $error</span><br>";
+		echo "<span style='color:red'><b>Upload Error</b> $fileType: $error</span><br>";
```
Comment (xlecours) [inline docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc:154]: don't use TABs, use 4 space characters

**Review by xlecours — CHANGES_REQUESTED** [review]

File: docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc (hunk 2 above)
Comment (laemtl) [reply docs/instruments/NDB_BVL_Instrument_UPLOADER_TEMPLATE.class.inc]: Oh, I understand. It's not a bad idea to add docs to this list.

**Pull request closed without merging**